documentation/architecture/12_DfE-Databricks.md

```diff
@@ -0,0 +1,124 @@
+
+#### Pros
+
+* **Consolidates All Data Sources:** Creates a single source of truth for all pipeline data (both database and static files). This simplifies data management, governance, and discovery.
+* **Simplified Ingestion Logic:** The FBIT pipeline only needs to know how to connect to one type of source system (Databricks), simplifying its ingestion code.
+
+#### Cons
+
+* **Delta lake datasets updates:** Who updates the ancillary data from gov.uk sources when it gets updated? Do we prompt the ADA team to make the new release available?
+* **Still Inefficient:** Suffers from the same major drawback as Option 1: all data is pulled out of Databricks for external processing, which is inefficient and costly.
+* **Marginal Improvement:** Offers only a small architectural improvement over Option 1 for a non-trivial amount of work.
+
+### **Option 3: Preprocess data in Databricks**
+
+Source data from and perform up until the gold layer of compute in Databricks. Calculate comparator sets in FBIT.
+
+#### Pros
+
+* **Leverages Databricks' Strengths:** Uses Databricks for what it excels at: large-scale, distributed data transformation (ETL/ELT). This makes the most performance-intensive part of the pipeline highly scalable. We'd also be more trackable via the unity catalogue.
+* **Fewer expensive data transfers:** The data pulled into FBIT is smaller and pre-aggregated (gold layer).
+* **Phased Modernisation:** Represents a good "best of both worlds" compromise. It modernises the core data processing while retaining specialised or complex logic ("comparator sets") in the existing, stable FBIT service.
+* **Aligns with Medallion Architecture:** This approach is a textbook implementation of the Bronze-Silver-Gold medallion architecture within a lakehouse platform which can be an example in the rest of DfE
+
+#### Cons
+
+* **Stray files not in databricks** We might need to still drag and drop new releases into databricks to make it all work, like we do with the current pipeline.
+* **Architectural Complexity:** The pipeline logic is now split across two systems (Databricks and FBIT). This can make orchestration, end-to-end monitoring, and debugging more complex.
+
+## Recommendation
+
+Option 2: Change all data sources we can to Databricks -lots of our current ancillary data is on there already.
+Low risk and easy. Not to rule out option 3 - When the databricks instance is used more and DfE have a vision for it to start handling transformation workloads we can start to think about migrating data pipelines on to it.
+For our multiple data sources we can migrate them at different times based on if they are in databricks:
+
+* Anything that is already in databricks we will look to clone extract logic asap - as new data drops come in through the year we can then validate that the migrated pipeline produces the same extract as the old pipeline and change it over.
```

Comment:
As discussed, there are 4 possibilities when we investigate each data drop:

1. the data are already in Databricks and we can obtain a version sufficiently similar to the current version
2. the data are not in Databricks but we can liaise with the ADA team such that they take on responsibility for ingesting and maintaining it as a dataset within Databricks
3. the data are not in Databricks but FBIT accepts responsibility for ingesting and maintaining it within Databricks
4. the data are not in Databricks, nor will we attempt to source it from Databricks (i.e. carry on as we are)